src/main/java/ch/njol/skript/events/EvtBlock.java

```diff
@@ -98,34 +106,47 @@ public boolean check(final Event e) {
 			return true;
 		
 		ItemType item;
+		BlockData blockData = null;
+
 		if (e instanceof BlockFormEvent) {
-			item = new ItemType(((BlockFormEvent) e).getNewState());
+			BlockFormEvent blockFormEvent = (BlockFormEvent) e;
+			BlockState newState = blockFormEvent.getNewState();
+			item = new ItemType(newState);
+			blockData = newState.getBlockData();
 		} else if (e instanceof BlockEvent) {
-			item = new ItemType(((BlockEvent) e).getBlock());
+			BlockEvent blockEvent = (BlockEvent) e;
+			Block block = blockEvent.getBlock();
+			item = new ItemType(block);
+			blockData = block.getBlockData();
 		} else if (e instanceof PlayerBucketFillEvent) {
-			item = new ItemType(((PlayerBucketEvent) e).getBlockClicked().getRelative(((PlayerBucketEvent) e).getBlockFace()));
+			PlayerBucketFillEvent playerBucketFillEvent = ((PlayerBucketFillEvent) e);
+			Block relative = playerBucketFillEvent.getBlockClicked().getRelative(playerBucketFillEvent.getBlockFace());
+			item = new ItemType(relative);
+			blockData = relative.getBlockData();
 		} else if (e instanceof PlayerBucketEmptyEvent) {
-			item = new ItemType(((PlayerBucketEmptyEvent) e).getItemStack());
-		} else if (Skript.isRunningMinecraft(1, 4, 3) && e instanceof HangingEvent) {
+			PlayerBucketEmptyEvent playerBucketEmptyEvent = ((PlayerBucketEmptyEvent) e);
+			item = new ItemType(playerBucketEmptyEvent.getItemStack());
+		} else if (e instanceof HangingEvent) {
 			final EntityData<?> d = EntityData.fromEntity(((HangingEvent) e).getEntity());
-			return types.check(e, new Checker<ItemType>() {
-				@Override
-				public boolean check(final @Nullable ItemType t) {
-					return t != null && Relation.EQUAL.is(DefaultComparators.entityItemComparator.compare(d, t));
-				}
+			return types.check(e, o -> {
+				if (o instanceof ItemType)
+					return Relation.EQUAL.is(DefaultComparators.entityItemComparator.compare(d, ((ItemType) o)));
```

Comment:
Does there need to be support for BlockData here?

Reply:
I'm 99.45% sure its a no, because a hanging event only involves an entity and an item.
No block involved here